internal/hwmon/hwmon.go

```diff
@@ -202,25 +189,25 @@ func GetFans(chip gosensors.Chip) map[int]*fans.HwMonFan {
 				min = fans.MinPwmValue
 			}
 
-			label := getLabel(chip.Path, inputSubFeature.Name)
+			label := getLabel(chip.Path, feature.Name)
 
 			fan := &fans.HwMonFan{
 				Config: configuration.FanConfig{
 					ID:     label,
 					MinPwm: &min,
 					MaxPwm: &max,
 					HwMon: &configuration.HwMonFanConfig{
-						Index:     currentOutputIndex,
-						PwmOutput: pwmOutput,
-						RpmInput:  rpmInput,
+						Index:      channel,
+						PwmChannel: channel,
+						SysfsPath:  chip.Path,
 					},
 				},
 				Label:        label,
-				Index:        currentOutputIndex,
+				Index:        channel,
 				RpmMovingAvg: rpmAverage,
 			}
 
-			result[currentOutputIndex] = fan
+			result[channel] = fan
```

Comment:
I cannot figure out how the code in this file is supposed to work. Can you please try to explain to me what the change is about and why you implemented in the way you did?

Reply:
My reply above explains the idea of a hwmon channel. The change to the hwmon controller is to map channel numbers to `Fan`s instead of enumeration indices.